SimTracker/SiPixelDigitizer/plugins/SiPixelDigitizerAlgorithm.cc

```diff
@@ -496,27 +496,62 @@ void SiPixelDigitizerAlgorithm::init_DynIneffDB(const edm::EventSetup& es, const
 }
 
 void SiPixelDigitizerAlgorithm::PixelEfficiencies::init_from_db(const edm::ESHandle<TrackerGeometry>& geom, const edm::ESHandle<SiPixelDynamicInefficiency>& SiPixelDynamicInefficiency) {
-
+  
   theInstLumiScaleFactor = SiPixelDynamicInefficiency->gettheInstLumiScaleFactor();
-  const std::map<uint32_t, double>& PixelGeomFactorsDB = SiPixelDynamicInefficiency->getPixelGeomFactors();
+  const std::map<uint32_t, double>& PixelGeomFactorsDBIn = SiPixelDynamicInefficiency->getPixelGeomFactors();
   const std::map<uint32_t, double>& ColGeomFactorsDB = SiPixelDynamicInefficiency->getColGeomFactors();
   const std::map<uint32_t, double>& ChipGeomFactorsDB = SiPixelDynamicInefficiency->getChipGeomFactors();
   const std::map<uint32_t, std::vector<double> >& PUFactors = SiPixelDynamicInefficiency->getPUFactors();
   std::vector<uint32_t > DetIdmasks = SiPixelDynamicInefficiency->getDetIdmasks();
   
-  // Loop on all modules, calculate geometrical scale factors and store in map for easy access
+  // Loop on all modules, initialize map for easy access
   for( const auto& it_module : geom->detUnits()) {
     if( dynamic_cast<PixelGeomDetUnit const*>(it_module)==nullptr) continue;
     const DetId detid = it_module->geographicalId();
     uint32_t rawid = detid.rawId();
     PixelGeomFactors[rawid] = 1;
     ColGeomFactors[rawid] = 1;
     ChipGeomFactors[rawid] = 1;
+    PixelGeomFactorsROCStdPixels[rawid] = std::vector<double>(16,1);
+    PixelGeomFactorsROCBigPixels[rawid] = std::vector<double>(16,1);
+  }
+  
+  // Store ROC level scale factors  (big and standard pixels separatelly)
+  std::map<uint32_t, double>  PixelGeomFactorsDB;
+  for (auto db_factor : PixelGeomFactorsDBIn){  
+    int shift = DetId(db_factor.first).subdetId() == 1? 6 : 3;    
+    unsigned int rocMask = 0x1F << shift;
+    unsigned int rocId = (((db_factor.first) & rocMask) >> shift);
+    if (rocId != 0) {
+      rocId--;
+      unsigned int rawid = db_factor.first & (~rocMask);      
+      double factor = db_factor.second;
+      double badFraction = 1 - factor;
+      double bigPixelFraction = 210./4160.;      
+      double stdPixelFraction = 3950./4160.;      
+      
```

Comment:
These numbers can be obtained by doing something like this somewhere where you have a concrete pixel DetId, or already a PixelGeomDetUnit:
const PixelGeomDetUnit * theGeomDet = dynamic_cast<const PixelGeomDetUnit*> (geom->idToDet(...a pixel det ID here...));
PixelTopology const * topology = &(theGeomDet->specificTopology());
and using topology->rowsperroc(), colsperroc(), nrows(), and ncolumns()